### PR TITLE
[workspace] deduplicate yarn lock in `postinstall`

### PR DESCRIPTION
**Pull request by Simek**
# Why

* https://github.com/atlassian/yarn-deduplicate#duplicated-packages

TL/DR: this script ensures that only the newest packages which matches semver are used in the workspace.

I have committed this to several repos and it has been used in (some major examples): 
* React Native core repo (no `postinstall` hook), 
* React Native website,
* React Native Directory,
* Docusuaurs (for all packages and websites, in repo for several months, no issues).

# How

This PR adds `yarn-deduplicate` to the workspace dependencies and hooks it up into `postinstall` check, before the Expo dependencies validation, to ensure that those checks are run on final lock version.

# Test Plan

`yarn build` on the workspace level was successful on `localhost`, let's try CI now.

CC: @brentvatne @EvanBacon

### CODE REVIEW

**Review by EvanBacon — APPROVED** [review]
seems fine
